### PR TITLE
Update tasks for rzls being packaged as a nuget

### PR DESCRIPTION
**Pull request by ryzngard**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -6,13 +6,9 @@
             "type": "extensionHost",
             "request": "launch",
             "runtimeExecutable": "${execPath}",
-            "args": [
-                "--extensionDevelopmentPath=${workspaceRoot}"
-            ],
+            "args": ["--extensionDevelopmentPath=${workspaceRoot}"],
```
Comment (ryzngard) [inline .vscode/launch.json:9]: Some of these changes came from the prettier json formatter. Idk if it matters, I just noticed some things being spaced differently and ran a formatter

File: .vscode/launch.json [hunk 2]
```diff
@@ -148,58 +118,36 @@
...
-            "NEW_DEPS_VERSION": "7.0.0-preview.24178.4",
-            "NEW_DEPS_ID": "RazorTelemetry"
-        },
-        "cwd": "${workspaceFolder}"
+            "type": "node",
+            "request": "launch",
+            "name": "Update razor telemetry package dependencies",
```
Comment (dibarbet) [inline .vscode/launch.json:133]: is there still work to be done to move Razor telemetry to the same workflow?

File: tasks/offlinePackagingTasks.ts [hunk 3]
```diff
@@ -167,6 +174,26 @@ gulp.task(
...
 );
 
+// Defines a special task to acquire all the platform specific Razor packages.
+// All packages need to be saved to the consumption AzDo artifacts feed, for non-platform
+// specific packages this only requires running the installDependencies tasks.  However for Razor packages
+// we need to acquire the nuget packages once for each platform to ensure they get saved to the feed.
+gulp.task(
+    'updateRazorVersion',
+    // Run the fetch of all packages, and then also installDependencies after
+    gulp.series(async () => {
```
Comment (dibarbet) [inline tasks/offlinePackagingTasks.ts:184]: this looks generally the same as `updateRoslynVersion`, consider sharing a function that passes the necessary component

File: package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
   "defaults": {
     "roslyn": "4.11.0-2.24267.3",
     "omniSharp": "1.39.11",
-    "razor": "7.0.0-preview.24266.1",
+    "razor": "9.0.0-preview.24311.4",
```
Comment (dibarbet) [inline package.json:42]: Can you also delete the entries in the package.json below that reference the razor bits from the blob storage?

File: .vscode/launch.json (hunk 2 above)
Comment (ryzngard) [reply .vscode/launch.json:133]: Yes. Working on that next

File: package.json (hunk 4 above)
Comment (ryzngard) [reply package.json:42]: oh yup! Missed that

**Review by dibarbet — APPROVED** [review]

**Comment by davidwengier** [comment]
Thank you <3

Can you update the instructions in https://github.com/dotnet/razor/blob/main/docs/InsertingRazorIntoVSCodeCSharp.md ? I legit don't actually know how to insert now :)

**Comment by ryzngard** [comment]
> Thank you <3
> 
> Can you update the instructions in [dotnet/razor@`main`/docs/InsertingRazorIntoVSCodeCSharp.md](https://github.com/dotnet/razor/blob/main/docs/InsertingRazorIntoVSCodeCSharp.md?rgh-link-date=2024-06-13T00%3A57%3A15Z) ? I legit don't actually know how to insert now :)

Yes! Good catch. I'm finishing up the telemetry insertion changes as well so I'll update once I put up the PR for that

**Pull request merged**